common/rpc/grpc.go

```diff
@@ -29,7 +29,13 @@ import (
 	"github.com/uber/cadence/common/service"
 )
 
-type GRPCPorts map[string]int
+type (
+	GRPCPortResolver interface {
+		GetGRPCAddress(service, hostAddress string) (string, error)
+	}
```

Comment:
Is this interface really needed?

If so, I would make it more general, as it does nothing specific to `grpc`

Reply:
The interface is needed, as integration tests pass different implementation.
But I agree that the name could be more general. Updated it.